sdks/go/pkg/beam/core/runtime/exec/fn_test.go

```diff
@@ -400,3 +400,210 @@ func BenchmarkInvokeFnCallExtra(b *testing.B) {
+	shimF := funcMakerFooRInt(wrFi)       // as if this shim were registered
+	shim1x1F := reflectx.ToFunc2x1(shimF) // would be MakeFunc1x1 if registered
+
+	var a int
+	var ai interface{} = a
+	aV := reflect.ValueOf(a)
+	rvSlice := []reflect.Value{aV}
+	grvSlice := []reflect.Value{gV, aV}
+	efaceSlice := []interface{}{a}
+	gEfaceSlice := []interface{}{g, a}
+
+	tests := []struct {
+		name string
+		fn   func()
+	}{
+		{"DirectMethod", func() { a = g.WhatsA(a) }}, // Baseline as low as we can go.
+		{"DirectFunc", func() { a = WhatsB(a) }},     // For comparison purposes
+
+		{"IndirectFunc", func() { a = indirectFunc(a) }},         // For comparison purposes
+		{"IndirectImplicit", func() { a = nrFi(a) }},             // Measures the indirection through reflect.Value cost.
+		{"TypeAssertedImplicit", func() { ai = nrFi(ai.(int)) }}, // Measures the type assertion cost over the above.
+
+		{"ReflectCallImplicit", func() { a = nrF.Call([]reflect.Value{reflect.ValueOf(a)})[0].Interface().(int) }},
+		{"ReflectCallImplicit-NoWrap", func() { a = nrF.Call([]reflect.Value{aV})[0].Interface().(int) }},
+		{"ReflectCallImplicit-NoReallocSlice", func() { a = nrF.Call(rvSlice)[0].Interface().(int) }},
+
+		{"ReflectXCallImplicit", func() { a = rxnrF.Call([]interface{}{a})[0].(int) }},
+		{"ReflectXCallImplicit-NoReallocSlice", func() { a = rxnrF.Call(efaceSlice)[0].(int) }},
+		{"ReflectXCall1x1Implicit", func() { a = rx0x1nrF.Call1x1(a).(int) }}, // Measures the default shimfunc overhead.
+
+		{"ShimedCallImplicit", func() { a = shimnrF.Call([]interface{}{a})[0].(int) }},          // What we're currently using for invoking methods
+		{"ShimedCallImplicit-NoReallocSlice", func() { a = shimnrF.Call(efaceSlice)[0].(int) }}, // Closer to what we're using now.
+		{"ShimedCall1x1Implicit", func() { a = shim0x1nrF.Call1x1(a).(int) }},
+
+		{"IndirectExplicit", func() { a = wrFi(g, a) }},                     // Measures the indirection through reflect.Value cost.
+		{"TypeAssertedExplicit", func() { ai = wrFi(gi.(*Foo), ai.(int)) }}, // Measures the type assertion cost over the above.
+
+		{"ReflectCallExplicit", func() { a = wrF.Call([]reflect.Value{reflect.ValueOf(g), reflect.ValueOf(a)})[0].Interface().(int) }},
+		{"ReflectCallExplicit-NoWrap", func() { a = wrF.Call([]reflect.Value{gV, aV})[0].Interface().(int) }},
+		{"ReflectCallExplicit-NoReallocSlice", func() { a = wrF.Call(grvSlice)[0].Interface().(int) }},
+
+		{"ReflectXCallExplicit", func() { a = rxF.Call([]interface{}{g, a})[0].(int) }},
+		{"ReflectXCallExplicit-NoReallocSlice", func() { a = rxF.Call(gEfaceSlice)[0].(int) }},
+		{"ReflectXCall2x1Explicit", func() { a = rx1x1F.Call2x1(g, a).(int) }},
+
+		{"ShimedCallExplicit", func() { a = shimF.Call([]interface{}{g, a})[0].(int) }},
+		{"ShimedCallExplicit-NoReallocSlice", func() { a = shimF.Call(gEfaceSlice)[0].(int) }},
+		{"ShimedCall2x1Explicit", func() { a = shim1x1F.Call2x1(g, a).(int) }},
+	}
+	for _, test := range tests {
```

Comment:
For my learning:

Would benchmarking output include a benchmark result for the BenchmarkMethodCalls too in this case? (Or how does b.Run allows it to distinguish that this is a wrapper not a direct benchmark.)

Reply:
The testing package GoDoc is canonical:
https://golang.org/pkg/testing/#hdr-Benchmarks
https://golang.org/pkg/testing/#hdr-Subtests_and_Sub_benchmarks

The short version is that what is invoked under b.Run and t.Run are for all intents and purposes distinct benchmarks and tests. It allows you to more concisely specify them with common code. Theres a side benefit in that instead of awkardly 
if failure {
  t.Error("asdfasdfas") 
  continue
}
continuing in table test loop, you can instead
if failure {
  t.Fatal("asdfasdfas") 
}
which won't abort subsequent tests that should be independant.